galore/cross_sections.py

```diff
@@ -281,3 +281,261 @@ def _eval_fit(energy, coeffs):
+    data['electron_counts'] = np.array(data['electron_counts']).astype(float)
+
+    # Build a dict_for_calculation which keys are 'PhotonEnergy' and orbitals like '1s1/2', '2s1/2', '2p1/2', '2p3/2', '3s1/2', '3p1/2', '3p3/2'...
+    # and values are connected cross sections and number of electrons of each orbital
+    # This is for calculating the max cross sections of 's', 'p', 'd', 'f' orbitals
+    new_lenth = len(data['electron_counts'])
+    cross_sections_by_orbitals = data['data_table'][-new_lenth:]
+    orbital_headers = data['headers'][-new_lenth:]
+
+    cross_sections_and_electron_numbers = np.concatenate(
+        (cross_sections_by_orbitals.T, [data['electron_counts']]), axis=0).T
+    dict_for_calculation = dict(
+        zip(orbital_headers, cross_sections_and_electron_numbers))
+    energy_array = data['data_table'][0]
+    dict_for_calculation['PhotonEnergy'] = energy_array
+
+    # match the import energy
+    index = np.where(dict_for_calculation['PhotonEnergy'] == energy)[0]
+
+    # build result dict
+    orbitals_cross_sections_dict = {}
+
+    # result for s orbital
+    s_cross_sections = np.array([dict_for_calculation[key]
+                                for key in dict_for_calculation if 's' in key]).T[index]
+    electrons_number = np.array([dict_for_calculation[key]
+                                for key in dict_for_calculation if 's' in key]).T[-1]
+    # get unit cross sections
+    unit_cross_sections = np.true_divide(s_cross_sections, electrons_number)
+    # get max cross section of obital s
+    max_cross_section = np.max(np.nan_to_num(unit_cross_sections))
+    orbitals_cross_sections_dict['s'] = max_cross_section
+
+    # result for 'p', 'd', 'f' orbitals
+    orbitals = ['p', 'd', 'f']
+    for orbital in orbitals:
+        interm_matrix = np.array([dict_for_calculation[key]
+                                 for key in dict_for_calculation if orbital in key]).T
+
+        if np.shape(interm_matrix) != (0,):
+            if reference == 'Scofield':
+                subshells_cross_sections = interm_matrix[index][0]
+
+                electrons_number = interm_matrix[-1]
+
+                result = get_avg_orbital_cross_sections(
+                    subshells_cross_sections, electrons_number)
+
+                # get max cross section of this obital
+                max_cross_section = np.max(np.nan_to_num(result))
+                orbitals_cross_sections_dict[orbital] = max_cross_section
+
+            elif reference == 'Yeh':
+                obital_cross_sections = interm_matrix[index][0]
+                electrons_number = interm_matrix[-1]
+                unit_cross_sections = np.true_divide(
+                    obital_cross_sections, electrons_number)
+                # get max cross section of this obital
+                max_cross_section = np.max(np.nan_to_num(unit_cross_sections))
+                orbitals_cross_sections_dict[orbital] = max_cross_section
```

Comment:
Again, we want the cross-section of the highest orbital, not the highest cross-section.